src/viperleed/gui/measure/widgets/pathselector.py

```diff
@@ -155,6 +157,14 @@ def set_path(self, new_path):
         """Set the contents of this path selector."""
         self.__glob['full_path'] = Path(new_path)
         self.__lineedit.setToolTip(str(new_path))
+        if self.__glob['select_file']:
+            valid = self.__glob['full_path'].is_file()
+            if not self.__glob['existing_file']:
+                valid = not valid
+        else:
+            valid = self.__glob['full_path'].is_dir()
```

Comment:
```suggestion
        full_path = self.__glob['full_path']
        selects_file = self.__glob['select_file']
        if not selects_file:
            valid = full_path.is_dir()
        elif selects_file and self.__glob['existing_file']:
            valid = full_path.is_file()
        else:  # A file to be selected for saving
            valid = True
```

A bit nicer to have a flat if...elif...else. Also, I think when a PathSelector is meant to select a file for saving, it does not matter whether the file exists or not. The assumption is that it will be created or overwritten.